R/utilities-units.R

```diff
@@ -347,14 +348,18 @@ getUnitsEnum <- function() {
         getUnitsForDimension(dimension = dimension)
       },
       error = function(cond) {
-        warning(messages$errorLoadingUnitsForDimension(dimension, cond))
+        errors <- c(errors, dimension)
         # making sure that in this case, the user sees that something went wrong
         return(c("Unavailable"))
       }
     )
     return(enum(replace(x, x == "", "Unitless")))
   })
 
+  if (length(errors) > 0L) {
```

Comment:
@IndrajeetPatil The errors vector is always empty. I am a bit confused here as I thought R would work with closure. What  am I doing wrong here. In this logic, I am collecting all errors at the function level and showing only one message. But it does not work.

Reply:
Hmm, at face value, this seems perfectly fine. I will see if I can figure out the issue locally.